index.js

```diff
@@ -3,22 +3,34 @@
 const fp = require('fastify-plugin')
 const elasticsearch = require('elasticsearch')
 
-function createClient (options) {
-  return new elasticsearch.Client(options)
-}
-
 function fastifyElasticSearch (fastify, options, next) {
-  var client = options.client || createClient(options)
+  const client = options.client || new elasticsearch.Client(options)
+  const requestTimeout = options.timeout
+
+  client.ping(
+    {
+      // ping usually has a 3000ms timeout
+      requestTimeout
+    },
+    function (err) {
+      if (err) {
+        next(err)
+      } else {
+        fastify.log.debug('elasticsearch cluster is available')
+        // plugin is ready
+        next()
+      }
+    }
+  )
 
   fastify
     .decorate('elasticsearch', client)
     .addHook('onClose', closeESClient)
 
-  next()
-}
-
-function closeESClient (fastify) {
-  return fastify.elasticsearch.close()
+  function closeESClient () {
+    fastify.log.debug('elasticsearch client is closing ...')
+    return client.close()
```

Comment:
Is this a sync function?
If not you need to handle the `done` callback.
https://github.com/mcollina/avvio#apponclosefunccontext-done

Reply:
It is sync https://github.com/elastic/elasticsearch-js/blob/14.x/src/lib/client.js#L54